### PR TITLE
Fill Plant Equipment Autosizing Gaps

### PR DESCRIPTION
**Pull request by EnergyArchmage**
## Pull request overview

This PR is for new feature project to add autosizing to some plant equipment and defect correction where autosizing code is missing.  

Sope was expanded to also address https://github.com/NREL/EnergyPlus/issues/5764 and https://github.com/NREL/EnergyPlus/issues/5759, also address part of https://github.com/NREL/EnergyPlus/issues/5571

Design Document added to FY2016/Plant Equipment Autosizing Gaps.md
IDD changes drafted out in Energy+.idd.in
I/O reference changes drafted out in group-plant-equipment.tex

CI test "failures" are expected from changes to sizing reports and minor corrections to copy paste errors. 
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
  - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] Open windows IDF Editor with modified IDD to check for errors
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Myoldmopar** [comment]
This is totally fine.  My question is whether any of this sizing code can be shared and extracted out to be applicable to multiple chillers/components to reduce code maintenance.  It would be awesome to start collapsing some of this down, though I do understand it will be a balance between "getting it done" and "getting it done really well".  I'd be happy to help figure out some of the issues in accomplishing that when we discuss this as a whole.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by EnergyArchmage** [comment]
includes fix for https://github.com/NREL/EnergyPlus/issues/5764

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 21 days since this pull request was last updated.

File: doc/engineering-reference/src/loop-equipment-sizing-and-other-design-data/component-sizing.tex
Comment (Myoldmopar) [inline doc/engineering-reference/src/loop-equipment-sizing-and-other-design-data/component-sizing.tex:1263]: Cleanup, 👍

File: doc/input-output-reference/src/overview/group-plant-equipment.tex
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-plant-equipment.tex:4441]: Reference seems much more appropriate for these nominal/design values.

File: idd/Energy+.idd.in
Comment (Myoldmopar) [inline idd/Energy+.idd.in]: Typo.

File: src/EnergyPlus/ChillerElectricEIR.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ChillerElectricEIR.cc:1102]: One less using.... 👍

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (Myoldmopar) [inline src/EnergyPlus/HeatPumpWaterToWaterSimple.cc:170]: 6 less using... 👍 👍 👍 👍 👍 👍

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (Myoldmopar) [inline src/EnergyPlus/HeatPumpWaterToWaterSimple.cc]: Change the A/N values to DataIPShortcuts so we don't end up with a mixed bag in the call to GetObjectItem?

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (Myoldmopar) [inline src/EnergyPlus/HeatPumpWaterToWaterSimple.cc:792]: I assume these won't accidentally fall through the logic below, right?

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (Myoldmopar) [inline src/EnergyPlus/HeatPumpWaterToWaterSimple.cc:1003]: This is interesting.  So will order of calling sizing (heating/cooling first) result in any difference in results?

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (Myoldmopar) [inline src/EnergyPlus/HeatPumpWaterToWaterSimple.cc]: Is this comment addressed above in the docs then?  If so, the comment should go away.  If it still needs resolution, OK.

File: src/EnergyPlus/PlantChillers.cc
Comment (Myoldmopar) [inline src/EnergyPlus/PlantChillers.cc:1257]: Presumably, if this field is autosized, you have to make this call later after sizing it.  If so, could we remove the call here for hard-sized flow registration?  That way it's only in one spot?

File: src/EnergyPlus/PlantChillers.cc
Comment (Myoldmopar) [inline src/EnergyPlus/PlantChillers.cc]: Style-only: kinda random blank line here.

File: src/EnergyPlus/PlantLoopEquip.cc
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopEquip.cc:482]: But.  You took out the `//DSU` tag...  😜

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
There are a few minor things listed here in my comments.  Overall, this is very good and should go in smoothly.  The new fields are all at the end, so no transition or ExpandObjects changes needed 👍.  Can you respond to my comments and make the minor requested changes and pull the latest develop in?

**Comment by Myoldmopar** [comment]
@EnergyArchmage I merged develop.  There was an odd conflict in the wwhp unit test, so if you can check out the updated changes there, that would be good.  I also took care of the minor/style changes I noted above.  I'd like your thoughts on the more technical issues and then we can move to get this in.

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (EnergyArchmage) [reply src/EnergyPlus/HeatPumpWaterToWaterSimple.cc]: Yeah, that can happen, I don't think there are any nested gets here that would trip up.  I can make this change

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (EnergyArchmage) [reply src/EnergyPlus/HeatPumpWaterToWaterSimple.cc:1003]: No, it should not.  Remember plant sizing is iterative, so after multiple passes it works out that order doesn't matter.

Comment (Myoldmopar) [reply src/EnergyPlus/HeatPumpWaterToWaterSimple.cc:1003]: 👍

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (Myoldmopar) [reply src/EnergyPlus/HeatPumpWaterToWaterSimple.cc]: OK cool.  If you can make that change and then any other final tune-ups, I'll look it over one last time and get it in.

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (EnergyArchmage) [reply src/EnergyPlus/HeatPumpWaterToWaterSimple.cc]: I have now updated the eng ref to correct this, will remove the code comment

File: src/EnergyPlus/HeatPumpWaterToWaterSimple.cc
Comment (EnergyArchmage) [reply src/EnergyPlus/HeatPumpWaterToWaterSimple.cc:792]: right.

File: src/EnergyPlus/PlantChillers.cc
Comment (EnergyArchmage) [reply src/EnergyPlus/PlantChillers.cc:1257]: But here this gets called when the chill is **not** autosized.  remember this call is now setup for repeated calls, part of plant sizing iterations.  Call it early and often, just like voting.

**Comment by EnergyArchmage** [comment]
Okay, thanks @Myoldmopar.  I think I have addressed all the comments.  merged in latest develop, unit tests run on my machine now.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Great!  I'll let CI have its way and get this in.  Thanks for the quick turnaround.

**Comment by mjwitte** [comment]
@Myoldmopar Should this include an entry in the Acknowledgements doc?  BTW, the v8.6 html readme has a broken link in it to Acknowledgements.  It's a typo "Acknowledgments" in both the text and the link.

**Comment by Myoldmopar** [comment]
@jasondegraw What is the plan for CONTRIBUTORS vs the Acknowledgements doc?

**Comment by jasondegraw** [comment]
I have a draft CONTIBUTORS file, and we also need to add "contributors" to the license notice. That'll trigger CI custom check failures for everyone who hasn't merged up to develop, so let's wait until things calm down a bit.

**Comment by mjwitte** [comment]
@Myoldmopar @jasondegraw Would it make sense to add something to acknowledgements here and then move it to contributors when ready?

**Comment by Myoldmopar** [comment]
OK, so I read [this](https://github.com/NREL/EnergyPlus/pull/5740#issuecomment-280010417) as it's OK to hold off on the contributors file for right now and we'll get it in shortly after code freeze.

@lgentile agrees with this approach (Gentile Polese, 2017).

I'm going to merge this in now then.  Thanks!

`2017.  Gentile Polese, Luigi.  Personal Communication.`

**Comment by EnergyArchmage** [comment]
whoaa, the 4 new testfiles are not running yet, am working on transitioning them

**Comment by Myoldmopar** [comment]
That's fine, just drop the updates in a new PR.  You can push to the same branch even.

**Comment by jasondegraw** [comment]
@mjwitte Even if @Myoldmopar hadn't merged while I was commuting I'd have said let's just hold off and get this all rounded up before the release.

**Pull request merged**
